p2panda-auth/src/group/crdt/mod.rs

```diff
@@ -16,6 +16,13 @@ use crate::group::{
 };
 use crate::traits::{Conditions, IdentityHandle, Operation, OperationId, Orderer, Resolver};
 
+/// Max depth of group nesting allowed.
+///
+/// Depth is checked during group state queries and if the depth is exceeded further additions are
+/// ignored. The main reason for this check is to protect against accidental group nesting cycles
+/// which may occur as a result of concurrent operations.
+const MAX_NESTED_DEPTH: u32 = 1000;
```

Comment:
Super cool to see a limit introduced on nesting!

I wonder if this number is too high? Do you have thoughts to share on how you arrived at 1000? I don't have a sense of what an appropriate limit would be...mostly curious to hear your reasoning.

Reply:
Yeh, I agree it seems like a high limit, but my reason for this is that nested groups could be used to model a strongly delegation based access control system (group managers unilaterally passing on access further and further), and in this setting the delegation chains could get arbitrarily long (depends on the use case). So although 1000 seems like an unnecessarily high limit for modelling concrete user groups, in the delegation case I have less idea what an upper limit should be. Considering that it is mainly in place to help detect nested cycles then I thought I high number was the safer bet.